### ISSUE TITLE
Support aws_autoscaling_group in keyvaluetags and with Provider ignore_tags Configuration

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#13808]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

All AWS services support resource tags with just a key and value, except Auto Scaling, where there is an additional `PropagateAtLaunch` field. The `keyvaluetags` package was deployed with provider `ignore_tags` support for all Terraform AWS Provider resources except `aws_autoscaling_group` due to this quirk. This issue serves to backfill that support into the last resource.

The first step will be adding "additional data" support into the `keyvaluetags` package functions, adding the `autoscaling` service name to all `keyvaluetags` generators, switching the `aws_autoscaling_group` resource `tag` attribute implementation to the generated handling, and adding ignore configuration support to the resource.

### Affected Resource(s)

* aws_autoscaling_group

### Potential Terraform Configuration

```hcl
provider "aws" {
  # ... potentially other configuration ...
  ignore_tags {
    keys = ["AmazonECSManaged"]
    key_prefixes = ["kubernetes.io/"]
  }
}
```

### References

- https://www.terraform.io/docs/providers/aws/#ignore_tags
- https://github.com/terraform-providers/terraform-provider-aws/issues/10688
- https://github.com/terraform-providers/terraform-provider-aws/issues/10689
- https://github.com/terraform-providers/terraform-provider-aws/pull/13039

**Comment by ewbankkit** [comment]
I started some work related to this in https://github.com/terraform-providers/terraform-provider-aws/pull/13508.

**Comment by bflad** [comment]
Support for this has been merged and will release with version 3.5.0 of the Terraform AWS Provider, later this week.
